src/rules/noAnyRule.ts

```diff
@@ -42,7 +43,10 @@ export class Rule extends Lint.Rules.AbstractRule {
 
 class NoAnyWalker extends Lint.RuleWalker {
     public visitAnyKeyword(node: ts.Node) {
-        this.addFailureAtNode(node, Rule.FAILURE_STRING);
+        const fix = this.createFix(
+            this.createReplacement(node.getStart(), node.getWidth(), "{}"),
```

Comment:
@pyrocat101 indeed, this linter autofix changes the (type) semantics of the code. I think that's the point of the linter warning though. You enable it when you don't want people to use `any` by default. As there is no semantically compatible option to `any` that we could automatically introduce, finding the appropriate replacement for `any` usually requires a human brain, so users most of the time will have to fix it themselves.

There is a large subset of cases though where users use `any` to mean `whatever`, i.e. a type with no requirements. Here, `{}` is an appropriate replacement.

I'd suggest either disabling the warning in the first place if you're cool with using `any` in your code base, or using the other auto-fix option that IDEs will show, i.e. adding the comment to disable the warning.